packages/block-library/src/verse/transforms.js

```diff
@@ -11,14 +11,44 @@ const transforms = {
 			transform: ( attributes ) =>
 				createBlock( 'core/verse', attributes ),
 		},
+		{
+			type: 'block',
+			blocks: [ 'core/quote' ],
+			isMatch: ( {}, blocks ) => {
+				// Only allow transform when quote contains only paragraphs or headings
+				const block = Array.isArray( blocks ) ? blocks[ 0 ] : blocks;
+				return block.innerBlocks.every(
+					( { name } ) =>
+						name === 'core/paragraph' || name === 'core/heading'
+				);
+			},
+			transform: ( {}, innerBlocks ) => {
+				// Extract content from the quote's inner blocks
+				const content = innerBlocks
+					.map( ( { attributes } ) => attributes.content || '' )
+					.filter( Boolean )
+					.join( '<br>' );
+				return createBlock( 'core/verse', { content } );
+			},
+		},
 	],
 	to: [
 		{
 			type: 'block',
 			blocks: [ 'core/paragraph' ],
+			priority: 10,
 			transform: ( attributes ) =>
 				createBlock( 'core/paragraph', attributes ),
 		},
+		{
+			type: 'block',
+			blocks: [ 'core/quote' ],
+			priority: 10,
+			transform: ( { content } ) =>
+				createBlock( 'core/quote', {}, [
+					createBlock( 'core/paragraph', { content } ),
+				] ),
+		},
```

Comment:
Transforming from a quote block to a quote block? We're already covering both quote > verse and verse > quote in the quote block transforms, so maybe this should just be removed anyway?